pallets/gear-debug/src/tests/mod.rs

```diff
@@ -927,11 +917,119 @@ fn check_gear_stack_end() {
                     state: crate::ProgramState::Active(crate::ProgramInfo {
                         static_pages: 4.into(),
                         persistent_pages,
-                        code_hash: h256_code_hash(&code),
+                        code_hash: utils::h256_code_hash(&code),
                     }),
                 }],
             })
             .into(),
         );
     })
 }
+
+#[test]
+fn disabled_program_rent() {
+    use test_syscalls::{Kind, WASM_BINARY as TEST_SYSCALLS_BINARY};
+
+    assert!(!<<Test as pallet_gear::Config>::ProgramRentEnabled as Get<bool>>::get());
+
+    init_logger();
+    new_test_ext().execute_with(|| {
+        let salt = b"salt".to_vec();
+        let pay_rent_id = ProgramId::generate(CodeId::generate(TEST_SYSCALLS_BINARY), &salt);
+
+        let program_value = 10_000_000;
+        assert_ok!(Gear::upload_program(
+            RuntimeOrigin::signed(1),
+            TEST_SYSCALLS_BINARY.to_vec(),
+            salt,
+            pay_rent_id.into_bytes().to_vec(),
+            20_000_000_000,
+            program_value,
+        ));
+
+        run_to_next_block(None);
+
+        let old_block = utils::get_active_program(pay_rent_id).expiration_block;
+
+        let block_count = 2_000u32;
+        let rent = RentCostPerBlockOf::<Test>::get() * u128::from(block_count);
+        let pay_rent_account_id = AccountId::from_origin(pay_rent_id.into_origin());
+        let balance_before = Balances::free_balance(pay_rent_account_id);
+        assert_ok!(Gear::send_message(
+            RuntimeOrigin::signed(1),
+            pay_rent_id,
+            vec![Kind::PayProgramRent(
+                pay_rent_id.into_origin().into(),
+                rent,
+                None
+            )]
+            .encode(),
+            20_000_000_000,
+            0,
+            false,
+        ));
+
+        run_to_next_block(None);
+
+        assert_eq!(utils::maybe_last_message(1).unwrap().payload_bytes(), b"ok");
+
+        // check that syscall does nothing
+        assert_eq!(
+            old_block,
+            utils::get_active_program(pay_rent_id).expiration_block
+        );
+        assert_eq!(balance_before, Balances::free_balance(pay_rent_account_id));
+
+        assert!(!ProgramStorageOf::<Test>::paused_program_exists(
```

Comment:
why should it be paused already? seems you wanted to assert pause task?

Reply:
assert checks that program is not paused yet